src/functional/program_processors/codegens/gtfn/gtfn_ir.py

```diff
@@ -76,12 +77,22 @@ class FunCall(Expr):
     args: list[Expr]
 
 
+class ScalarFunCall(Expr):
+    fun: SymRef
+    args: list[Expr]
+
+
 class FunctionDefinition(Node, SymbolTableTrait):
     id: Coerced[SymbolName]  # noqa: A003
     params: list[Sym]
     expr: Expr
 
 
+class ScalarFunDef(Node, SymbolTableTrait):
+    id: Coerced[SymbolName]
```

Comment:
```suggestion
    id: Coerced[SymbolName]   # noqa: A003
```

I think this will be needed here